trpcultivate_germcollection/src/Plugin/TripalImporter/GermplasmRelationshipImporter.php

```diff
@@ -0,0 +1,1189 @@
+              // validator plugin, then only set the organism ID if the organism
+              // is valid.
+              // Organism ID:
+              // If Scientific Name is present, lookup the organism ID
+              // and set it in the validator.
+              if ($data_row[2]) {
+                $organism_id = $this->getOrganismIds($data_row[2]);
+              }
+              else {
+                $organism_id = NULL;
+              }
+              if ($organism_id == NULL) {
+                // Manually set failures array.
+                $failed_validator = TRUE;
+                $failures[$validator_name][$line_no] = [
+                  'case' => 'Unable to lookup germplasm with empty values',
+                  'valid' => FALSE,
+                  'failedItems' => ['empty_cells' => $line_no],
+                ];
+                continue;
+              }
+              $validator->setOrganismID($this->organism_ids[$data_row[2]]);
+            }
+            $result = $validator->validateRow($data_row);
+            // Check if validation failed.
+            if (array_key_exists('valid', $result) && $result['valid'] === FALSE) {
+              $failed_validator = TRUE;
+              $failures[$validator_name][$line_no] = $result;
+            }
+          }
+        }
+      }
+
+      fclose($handle);
+    }
+
+    $validation_feedback = $this->processValidationMessages($failures);
+    $storage = $form_state->getStorage();
+    $storage[self::VALIDATION_RESULT] = $validation_feedback;
+    $form_state->setStorage($storage);
+
+    $submit_form = TRUE;
+
+    foreach ($validation_feedback as $feedback_item) {
+      if (($feedback_item['status'] == 'todo' && $form_values['relationship_toggle']) || $feedback_item['status'] == 'fail') {
+        $submit_form = FALSE;
+        break;
+      }
+    }
+
+    if ($submit_form === FALSE) {
+      $this->service_Messenger
+        ->addError('Your file import was not successful. Please check the Validation Result Window for errors and try again.');
+
+      $form_state->setRebuild(TRUE);
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function form($form, &$form_state) {
+
+    $form = parent::form($form, $form_state);
+
+    // INFO:
+    $info = $this->t('This importer will create individuals of a population and
+        relate them back to the population stock. More specifically, for every line
+        in the file, a new chado stock record with that information will be created.
+        Then a relationship as specified in this form will be made between that new
+        stock record and the population stock selected in this form. As such this
+        importer can be used in any case where you want to create a number of new
+        stock records related to an existing stock. Examples of such situations are
+        recombinant inbred line populations or nested association mapping panels.');
+    $this->service_Messenger->addStatus($info);
```

Comment:
This is now outdated for this importer, plus we'll want to avoid using the word "stock" when it's something user-facing. Here's my suggestion:
```suggestion
    $info = $this->t('This importer will relate a group of germplasm to an existing
        germplasm. More specifically, for every line in the file, the importer will
        look up to see if the germplasm already exists or create the germplasm if
        allowed to do so. Then, a relationship, with the type specified in this
        form, will be made between that new germpasm and the primary germplasm
        selected in this form. An example use case of this importer is to relate
        multiple individuals to a single breeding cross.');
    $this->service_Messenger->addStatus($info);
```